src/tbb/misc_ex.cpp

```diff
@@ -76,7 +78,7 @@ static void set_thread_affinity_mask( std::size_t maskSize, const basic_mask_t*
 }
 
 static void get_thread_affinity_mask( std::size_t maskSize, basic_mask_t* threadMask ) {
-#if __linux__
+#if __unix__
     if( sched_getaffinity( 0, maskSize, threadMask ) )
 #else /* FreeBSD */
     if( cpuset_getaffinity( CPU_LEVEL_WHICH, CPU_WHICH_TID, -1, maskSize, threadMask ) )
```

Comment:
AFAIK, FreeBSD does not support `sched_getaffinity` but defines `__unix__`. Does it really works on arbitrary Unix? The affinity API is not part of POSIX.

Reply:
As said in the previous comment, FreeBSD does not define unix.

Reply:
@pablorcum 
According to [this page](https://forums.freebsd.org/threads/gcc-general-macro-for-bsd.47769/), the `__unix__` macro is defined on FreeBSD systems. Unfortunately, I have no available FreeBSD machine to test this statement.

Maybe the `__linux__ && __QNXNTO__` condition can be used instead of `__unix__`?

Reply:
Yes, that's also possible. I don't have anything against that. My personal preference is to keep it as general and reusable as possible, therefore I tend to use __unix__ instead of __linux__ in my other projects, but we can keep __linux__ || __QNXNTO__ for the sake of backwards compatibility.

Reply:
@ivankochin , are you sure that it is a good idea to limit the patch for QNX only? I suppose enabling Unix in generic way and implementing some exclusions (e.g. for FreeBSD) is the good approach for other Unix-like platforms.

Reply:
@alexey-katranov, Ok, then we may use a common approach with `__unix__` everywhere except the affinity interfaces case. Do you know more differences between FreeBSD and common POSIX interfaces?

Reply:
As I understand, there is no affinity interfaces in POSIX. So, for each platform (or group of platforms) it should be implemented in its own way (e.g. disabled if possible).

Reply:
Swap Unix and *BSD blocks:
```suggestion
#if __FreeBSD__ || __NetBSD__ || __OpenBSD__
    if( cpuset_getaffinity( CPU_LEVEL_WHICH, CPU_WHICH_TID, -1, maskSize, threadMask ) )
#else /* __unix__ */
    if( sched_getaffinity( 0, maskSize, threadMask ) )
```